Cauldron/Assets/Scripts/RecipeBook.cs

```diff
@@ -56,6 +56,8 @@ public class RecipeBook : Book, IDataPersistence
         public event Action OnSelectIncorrectRecipe;
         public event Action OnOpenBook;
         public event Action OnUnlockAutoCooking;
+        public event Action<bool> AvailableChange;
+        public event Action<string, int> ChangeBookButtonLayer;
```

Comment:
And this one looks super weird)
Maybe it's better to Register the book button in the recipe book as HudButton or something (in Awake) and then call the button methods directly.